pkg/reconciler/knativeeventing/common/images.go

```diff
@@ -77,16 +78,19 @@ func updateDeploymentImage(deployment *appsv1.Deployment, registry *eventingv1al
 	containers := deployment.Spec.Template.Spec.Containers
 	for index := range containers {
 		container := &containers[index]
-		newImage := getNewImage(registry, container.Name)
+		newImage := getNewImage(registry, container.Name, deployment.Name)
 		if newImage != "" {
 			updateContainer(container, newImage, log)
 		}
 	}
 	log.Debugw("Finished updating images", "name", deployment.GetName(), "containers", deployment.Spec.Template.Spec.Containers)
 }
 
-func getNewImage(registry *eventingv1alpha1.Registry, containerName string) string {
-	overrideImage := registry.Override[containerName]
+func getNewImage(registry *eventingv1alpha1.Registry, containerName, deploymentName string) string {
+	overrideImage := registry.Override[deploymentName+delimiter+containerName]
+	if overrideImage == "" {
```

Comment:
please add a test where override for "deploymentName+delimiter+containerName" and "containerName" is specified, and that "deploymentName+delimiter+containerName" wins out, that is to verify that deploymentName takes precedence over just containerName

Reply:
That is what exactly the test cases in image_test.go is doing:
```
OverrideWithDeploymentContainer, OverridePartialWithDeploymentContainer, OverrideWithDeploymentName
```

They respectively check overriding both of two containers, overriding one container, and using `deploymentName+delimiter+containerName` only.